src/mps/mpo.jl

```diff
@@ -258,3 +293,66 @@ function densityMatrixApplyMPO(A::MPO, psi::MPS; kwargs...)::MPS
     psi_out.rlim_ = 2
     return psi_out
 end
+
+function nmultMPO(A::MPO, B::MPO; kwargs...)::MPO
+    cutoff::Float64 = get(kwargs, :cutoff, 1e-14)
+    resp_degen::Bool = get(kwargs, :respect_degenerate, true) 
+    maxdim::Int = get(kwargs,:maxdim,max(maxDim(A), maxDim(B)))
+    mindim::Int = max(get(kwargs,:mindim,1), 1)
+    n = length(A)
+    n != length(B) && throw(DimensionMismatch("lengths of MPOs A ($n) and B ($(length(B))) do not match"))
+    A_ = copy(A)
+    position!(A_, 1)
+    B_ = copy(B)
+    position!(B_, 1)
+
+    links_A = findinds.(A.A_, "Link")
+    links_B = findinds.(B.A_, "Link")
+
+    for i in 1:n
+        if length(commoninds(findinds(A_[i], "Site"), findinds(B_[i], "Site"))) == 2
+            A_[i] = prime(A_[i], "Site")
+        end
+    end
+    res = similar(A_)
+    # otherwise we will end up modifying the elements of A!
+    res.A_ = deepcopy(A_.A_)
+    # need to reindex link indices of res
+    for i in 1:n-1
+        ci = commonindex(res[i], res[i+1])
+        new_ci = Index(dim(ci), tags(ci))
+        replaceindex!(res[i], ci, new_ci)
+        replaceindex!(res[i+1], ci, new_ci)
+        @assert commonindex(res[i], res[i+1]) != commonindex(A[i], A[i+1])
+    end
+    sites_A = [setdiff(findinds(x, "Site"), findindex(y, "Site"))[1] for (x,y) in zip(A_.A_, B_.A_)]
+    sites_B = [setdiff(findinds(x, "Site"), findindex(y, "Site"))[1] for (x,y) in zip(B_.A_, A_.A_)]
+    res[1] = ITensor(sites_A[1], sites_B[1], commonindex(res[1], res[2]))
+    for i in 1:n-2
+        if i == 1
+            clust = A_[i] * B_[i]
+        else
+            clust = nfork * A_[i] * B_[i]
+        end
+        lA = commonindex(A_.A_[i], A_.A_[i+1])
```

Comment:
Is getindex not defined for MPO? If so I think we should define and use it here: `A_[i]` instead of `A_.A_[i]`